### DISCUSSION TITLE
Problems getting data from params

### DISCUSSION THREAD

**Discussion by gabrielsze** [nuxt/nuxt#9753] [Questions (Nuxt 2)] [answered]
Hi all, I have this weird problem I ran into.

Let's assume I have two brands, maybe brand 1 and brand-2 (one name with spacing, and the other hyphenated by nature). I am trying to generate _brand.vue pages.

I am trying to use a hyphenated link structure in my pages, meaning

example.com/brands/brand-1
example.com/brands/brand-2

so i end up using a regex to replace all spaces with hyphens.
```
let brandHyphen = brand.replace(/\s+/g, "-");
return `/brands/${brandHyphen}`
```

However, inside my `asyncData` of the pages, when i am trying to generate and I grab params.brand, I apply a regex to replace the hyphens with space (in hopes to get back the original brand name), I get back brand 1 (correct) and brand 2 (wrong).

How can I retain the hyphenated brand names in this process, or are there any ways around this?

This is how my asyncData looks like inside `_brand.vue`

```
  async asyncData({ params, payload }) {
    const brand = params.brand; // grabbing brand from params gives me brand-1 and brand-2

    // runs some regex to replace hyphens with spaces
    brand = brand.replace(/-/g, ' ');

    // brand 1 is correct
    // brand 2 is wrong (should be brand-2)

    // try to use brand data
    array.filter(x => x === brand); // finds brand 1 but can't find brand-2
  }
```

**Comment by TheAlexLichter** [comment]
Please share some code ☺️

**Reply by gabrielsze** [reply]
I have updated some small snippets but not sure what else to share, haha.

**Reply by JHerwig** [reply]
Your code snippet contains only an empty function declaration. Maybe your regex is wrong. But therefore we need to see your code behind the comments.

**Reply by gabrielsze** [reply]
Thanks for the heads up - i have updated the regex that i used.

**Comment by JHerwig** [comment, marked as answer]
Your first expression `brand.replace(/\s+/g, "-")` seems fine. What exactly doesn't work? 
What form is `brand` look like? Do you pass each value at once or one by another?

**Reply by gabrielsze** [reply]
Hey, i think the main issue is that some brands already have pre-existing hyphens inside of them, and once i apply a regex (to replace spaces with hyphens), I can't seem to get back their original brand names from the URL

**Reply by JHerwig** [reply]
First off, that is nothing related to nuxt.
Nevertheless, that is true, once you request  `/brands/my-cool-brand-name` you can't decide what was the original name. (As a hint, you could use escaping, but that doesn't beautify your URL or using some other char than '-')
Besides, it could be a problem if there are two brands with
`cool-name` and `cool name` which both get mapped to the same URL. 

Is the original problem solved? Then please close this thread.
